File: lista.cpp
#include <stdio.h>

#define SIZE 10


void printArray(int arr[], int size) {
    for (int i = 0; i < size; i++) {
        printf("numbers[%d] = %d\n", i, arr[i]);
    }
}

int main() {
    int numbers[SIZE] = {1, 2, 3, -4, 50, 6, 7, 8, 9, 10};
    int sorted[SIZE];
    int sum = 0, minIdx = 0, maxIdx = 0;

    printf("Original array:\n");
    printArray(numbers, SIZE);


    for (int i = 0; i < SIZE; i++) {
        sum += numbers[i];
        if (numbers[i] < numbers[minIdx]) minIdx = i;
        if (numbers[i] > numbers[maxIdx]) maxIdx = i;
    }

    printf("\nMaximum: %d\n", numbers[maxIdx]);
    printf("Minimum: %d\n", numbers[minIdx]);
    printf("Sum: %d\n", sum);
    printf("Average: %.2f\n", (float)sum / SIZE);


    int temp = numbers[minIdx];
    numbers[minIdx] = numbers[maxIdx];
    numbers[maxIdx] = temp;

    printf("\nArray after swapping minimum and maximum:\n");
    printArray(numbers, SIZE);


    for (int i = 0; i < SIZE; i++) {
        sorted[i] = numbers[i];
    }


    for (int i = 0; i < SIZE - 1; i++) {
        for (int j = 0; j < SIZE - 1 - i; j++) {
            if (sorted[j] > sorted[j + 1]) {
                temp = sorted[j];
                sorted[j] = sorted[j + 1];
                sorted[j + 1] = temp;
            }
        }
    }

    printf("\nSorted array:\n");
    printArray(sorted, SIZE);


    float median = (SIZE % 2 == 0)
        ? (sorted[SIZE / 2 - 1] + sorted[SIZE / 2]) / 2.0
        : sorted[SIZE / 2];

    printf("\nMedian: %.2f\n", median);

    return 0;
}

File: lista.py
SIZE = 10


def print_array(arr):
    for i in range(len(arr)):
        print(f"numbers[{i}] = {arr[i]}")


def main():
    numbers = [1, 2, 3, -4, 50, 6, 7, 8, 9, 10]

    print("Original array:")
    print_array(numbers)


    sum_val = 0
    min_idx = 0
    max_idx = 0


   

    for i in range(SIZE):
        sum_val += numbers[i]
        if numbers[i] < numbers[min_idx]:
            min_idx = i
        if numbers[i] > numbers[max_idx]:
            max_idx = i

    print(f"\nMaximum: {numbers[max_idx]}")
    print(f"Minimum: {numbers[min_idx]}")
    print(f"Sum: {sum_val}")
    print(f"Average: {sum_val / SIZE:.2f}")


    numbers[min_idx], numbers[max_idx] = numbers[max_idx], numbers[min_idx]

    print("\nArray after swapping minimum and maximum:")
    print_array(numbers)


    sorted_arr = numbers.copy()

    #
    for i in range(SIZE - 1):
        for j in range(SIZE - 1 - i):
            if sorted_arr[j] > sorted_arr[j + 1]:

                sorted_arr[j], sorted_arr[j + 1] = sorted_arr[j + 1], sorted_arr[j]

    print("\nSorted array:")
    print_array(sorted_arr)

    if SIZE % 2 == 0:
        median = (sorted_arr[SIZE // 2 - 1] + sorted_arr[SIZE // 2]) / 2.0
    else:
        median = sorted_arr[SIZE // 2]

    print(f"\nMedian: {median:.2f}")

if __name__ == "__main__":
    main()